presto-main/src/test/java/com/facebook/presto/sql/planner/BenchmarkPlanner.java

```diff
@@ -0,0 +1,161 @@
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+import org.openjdk.jmh.runner.options.WarmupMode;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Warmup(iterations = 5)
+@Fork(1)
+@Measurement(iterations = 20)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkPlanner
+{
+    @SuppressWarnings("FieldMayBeFinal")
+    @State(Scope.Benchmark)
+    public static class BenchmarkData
+    {
+        @Param({"", "EXPLAIN"})
+        private String queryPrefix = "";
+
+        @Param({"true", "false"})
+        private String iterativeOptimizerEnabled = "true";
+
+        private LocalQueryRunner queryRunner;
+        private List<String> queries;
+        private Session session;
+
+        @Setup
+        public void setup()
+        {
+            String tpch = "tpch";
+
+            session = testSessionBuilder()
+                    .setCatalog(tpch)
+                    .setSchema("sf1000")
+                    .setSystemProperty("iterative_optimizer_enabled", iterativeOptimizerEnabled)
+                    .build();
+
+            queryRunner = new LocalQueryRunner(session);
+            queryRunner.createCatalog(tpch, new TpchConnectorFactory(4), ImmutableMap.of("tpch.column-naming", "standard"));
+
+            queries = IntStream.rangeClosed(1, 22)
+                    .boxed()
+                    .filter(i -> i != 15) // q15 has two queries in it
+                    .map(i -> readResource(String.format("/io/airlift/tpch/queries/q%d.sql", i)))
+                    .collect(toImmutableList());
+        }
+
+        @TearDown
+        public void tearDown()
+        {
+            queryRunner.close();
+            queryRunner = null;
+        }
+
+        public String readResource(String resource)
+        {
+            try {
+                URL resourceUrl = Customer.class.getResource(resource);
+                return Resources.toString(resourceUrl, Charsets.UTF_8);
+            }
+            catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        public Stream<String> getQueries()
+        {
+            return queries.stream()
+                    .map(query -> queryPrefix + "\n" + query);
+        }
+    }
+
+    @Benchmark
+    public List<Plan> planQueries(BenchmarkData benchmarkData)
+    {
+        return benchmarkData.queryRunner.inTransaction(txSession -> benchmarkData.getQueries()
```

Comment:
`s/txSession/transactionSession`

conver lambda to `{ .. }` form

Reply:
Anything which makes you happy.